packages/mdc-drawer/dismissible/foundation.js

```diff
@@ -55,9 +54,13 @@ class MDCDismissibleDrawerFoundation extends MDCFoundation {
     if (this.isOpen() || this.isOpening() || this.isClosing()) {
       return;
     }
+
     this.adapter_.addClass(cssClasses.OPEN);
-    this.adapter_.addClass(cssClasses.ANIMATING_OPEN);
-    this.animateAppContent_(true);
+    this.adapter_.addClass(cssClasses.ANIMATE);
+
+    // Runs this after Style and Layout information have been calculated and the paint has occurred.
+    requestAnimationFrame(() => setTimeout(() => this.adapter_.addClass(cssClasses.OPENING), 0));
```

Comment:
The idea with rAF was that setTimeout wouldn't be needed... is it still needed here? Or can we force recalc and not require waiting at all? (e.g. in https://github.com/material-components/material-components-web/blob/master/packages/mdc-tab-indicator/sliding-foundation.js)

Reply:
Please see my [comment](https://github.com/material-components/material-components-web/pull/3340/files#r210674599) below.

Firefox browser executes the rAF callback before layout / paint where but works on Chrome without `setTimeout`. Causing force reflow works but wanted to additional reflow.

Reply:
Is my idea really adding additional reflow, or is it just making the same reflow happen sooner? If it works it would simplify test cases and possibly also reduce potential for issues in framework wrappers...

Reply:
I suspect Matt is coming from the perspective of MDC React where destroying something while a rAF/setTimeout is still pending might cause issues unless they tiptoe around `addClass` to guard against it being called after/during destroy.

It might help if we stored references to the scheduled rAF and setTimeout and cleared them in destroy.

Reply:
As per MDN documentation it says rAF with setTimeout set to `0` causes it to execute immediately after previous layout + paint is complete.

It would be helpful to look at an example where it might cause an issue.

Reply:
Found an edge case where using rAF + setTimeout might be an issue.

    drawer.open // true
    drawer.isOpening // false - because rAF callback is still pending

This wouldn't happen if we force reflow. Added new foundation method `computeBoundingRect` for reflow. PTAL.

Thanks!